mbuild/tests/test_utils.py

```diff
@@ -35,3 +36,30 @@ def test_import(self):
 
         with pytest.raises(ImportError):
             import_('garbagepackagename')
+
+    def test_js_utils(self):
+        nglview = import_('nglview')
+        with pytest.raises(TypeError):
+            nglview_custom_tooltip(object())
+        test_widget = nglview.NGLWidget()
+        nglview_custom_tooltip(test_widget)
+        assert hasattr(test_widget, 'stage')
+        assert isinstance(test_widget._ngl_msg_archive, list)
+        assert len(test_widget._ngl_msg_archive) == 1
+        assert isinstance(test_widget._ngl_msg_archive[0], dict)
+        message_dict = test_widget._ngl_msg_archive[0]
+        assert message_dict['target'] == 'Widget'
+        assert message_dict['type'] == 'call_method'
+        assert message_dict['methodName'] == 'executeCode'
+        assert message_dict['args'] == [
+                    """
+                    this.stage.mouseControls.add('hoverPick', (stage, pickingProxy) => {
+                        let tooltip = this.stage.tooltip;
+                        if(pickingProxy && pickingProxy.atom && !pickingProxy.bond){
```

Comment:
Just to satisfy my curiosity, if we support more features from nglview, would this `args` section essentially be extended?

Reply:
Yeah. It depends on the type of support `nglview` has in the future. If they have additional python methods in the future, we wouldn't have to rely on this `js` call. There are a lot of features in `ngl` which cannot be implemented directly from `nglview` without firing `js`.